live-build/misc/upgrade-scripts/verify-impl

```diff
@@ -63,14 +63,30 @@ if $opt_d; then
 	VERIFY_LIVE_MDS_OPT="-disableConsistentMdsZfsDataUtil"
 fi
 
-trap cleanup EXIT
+#
+# We use this directory to store the upgrade verification logs of this
+# container. This directory is accessible from both the container, and
+# the host. This allows the software running on the host to easily
+# access the logs after the verification is performed.
+#
+# Additionally, we only want to save the logs for one verification
+# invocation at any given time; mostly to avoid these logs from
+# consuming storage unnecessarily. Thus, we delete the existing
+# directory first, and then recreate an empty directory.
+#
+rm -rf "${LOG_DIR:?}" ||
+	die "failed to remove directory: '${LOG_DIR}'"
+mkdir -p "${LOG_DIR}/${CONTAINER}" ||
+	die "failed to create directory: '${LOG_DIR}/${CONTAINER}'"
+
+trap run_support_info_script EXIT
 
 /usr/bin/java \
 	-Dlog.dir=/var/delphix/server/upgrade-verify \
 	-Dmdsverify=true \
 	$VERIFY_DEBUG_OPT \
 	-jar /opt/delphix/server/lib/exec/upgrade-verify/upgrade-verify.jar \
-	-d "${opt_o:-$DROPBOX_DIR/upgrade_verify_report.json}" \
+	-d "${opt_o:-${LOG_DIR}/${CONTAINER}/upgrade_verify_report.json}" \
```

Comment:
IMO, what I'd like us to do is to convert the logic that we currently do in the `verify` script, and rewrite that logic in our Java stack; and then remove the `verify` script. If we did that, then the Java logic would obtain the container's name after it creates the container (just like how `verify` gets the container name). I think this is the way we should approach it, because I think we're going to need this logic in Java to support verification for "not-in-place" upgrades, so it'd be good if both upgrade types shared similar code for verification.

If we don't go that route, then I see a few potential solutions:

1. Use `verify-impl`'s default; I think this is reasonable, especially considering we don't have an option for specifying where the `support_info` output is placed. So, if we used the default for the JSON file, it'd be consistent with what we're doing for `support_info`.

2. Have the app-stack specify `/var/dlpx-update/verify-logs`, and then we add the container name to that path. If we did it this way, the Java logic would have to assume only one container directory would exist after verification is complete, otherwise how would it choose which container directory was the one it wants? Presumably it already makes (or will make) this assumption when consuming the `support_info` output.

3. Don't use a container-specific directory at all. I know I pushed to do this originally, but perhaps this is just making things for difficult for little benefit?

If we don't want to remove `verify` and re-implement that logic in Java, then I'd probably prefer to go with (2) or (3), and also update the script to also place the `support_info` output in the directory specified with `-d` to be consistent.

Reply:
I agree that porting `verify` script logic to Java stack is the best solution here.

Reply:
Do we need a fix for this ASAP? I think, as it stands now, this JSON file isn't going to be accessible from outside of the container.

Is this breakage OK in the short term, and we can work on porting the `verify` logic to Java? Or should we get a fix in now, even though we might revert that fix later?